interface/main/holidays/import_holidays.php

```diff
@@ -0,0 +1,192 @@
+require_once("$srcdir/sql.inc");
+require_once("Holidays_Controller.php");
+
+if (!acl_check('admin', 'super')) die(xlt('Not authorized'));
+
+// Handle uploads.
+$holidays_controller = new Holidays_Controller();
+$csv_file_data = $holidays_controller->get_file_csv_data();
+
+if (!empty($_POST['bn_upload'])) {
+    //Upload and save the csv
+    $saved = $holidays_controller->upload_csv($_FILES);
+}
+if (!empty($_POST['import_holidays'])) {
+    //Import from the csv file to the calendar external table
+    $saved = $holidays_controller->import_holidays_from_csv();
+}
+if (!empty($_POST['sync'])) {
+    //Upload and save the csv
+    $saved = $holidays_controller->create_holiday_event();
+}
+
+
+?>
+
+<html>
+<head>
+    <title><?php echo xlt('Holidays management'); ?></title>
+    <link rel="stylesheet" href='<?php echo $css_header ?>' type='text/css'>
+
+</head>
+
+<body class="body_top">
+<?php
+if ($saved){
+    echo "<p style='color:green'>" .
+        xlt('Successfully Completed');
+        "</p>\n";
+}elseif(!empty($_POST['bn_upload'])             &&
+        !empty($_POST['import_holidays'])       &&
+        !empty($_POST['sync'])
+        ){
+    echo "<p style='color:red'>" .
+        xlt('Operation Failed' );
+    "</p>\n";
+}
+?>
+<form method='post' action='import_holidays.php' enctype='multipart/form-data'
+      onsubmit='return top.restoreSession()'>
+
+        <p class='text'>
+        <table border='1' cellpadding='4'>
+            <tr bgcolor='#dddddd' class='dehead'>
+                <td align='center' colspan='2'>
+                    <?php echo xlt('CSV'); ?>
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo htmlspecialchars(xl('CSV File')); ?>
+                    <input type="hidden" name="MAX_FILE_SIZE" value="350000000" />
+                </td>
+                <td class='detail' nowrap>
+                    <input type="file" name="form_file" size="40" />
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo xlt('File on server (modification date)'); ?>
+                </td>
+                <td class='detail' nowrap>
+                    <?php
+                    if(!empty($csv_file_data)){?>
+
+                        <?php $path=explode("/",$holidays_controller->get_target_file());?>
+                        <?php $filename=$path[count($path)-1];?>
+                        <?php unset($path[count($path)-1]);?>
+
+                        <?php
+                        // build the path to the for the download request
+                        $phppagepath = "http://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]?filename=";?>
+                        <a href="#" onclick='window.open("<?php echo $phppagepath; echo $holidays_controller->get_target_file();?>&download_file=1")'><?php echo text($csv_file_data['date']);?></a>
```

Comment:
As discussed above:
The target file is hard-coded, so no reason to bring it in as a get parameter. the download portion of the script can collect without passing it as a get parameter.